lang/csharp/src/apache/main/Schema/UnionSchema.cs

```diff
@@ -161,5 +164,17 @@ public override int GetHashCode()
             result += getHashCode(Props);
             return result;
         }
+
+        private void VerifyChildSchemas(List<Schema> schemas)
+        {
+            if (schemas.Any(schema => schema.Tag == Type.Union))
+                throw new ArgumentException("Unions may not immediately contain other unions", nameof(schemas));
+
+            var groupedByFullNames = schemas.GroupBy(schema => schema.Fullname);
+            IGrouping<string, Schema> duplicateType = groupedByFullNames.FirstOrDefault(x => x.Count() > 1);
+
+            if (duplicateType != null)
```

Comment:
You can move your linq statement into the if.  There is no reason to create a variable that is not used in another place.

Reply:
It cannot be moved, I am calculating 'duplicateType' in the linq and checking it's value in the if statement.

Reply:
```
IGrouping<string, Schema> duplicateType = schemas.GroupBy(schema => schema.Fullname).FirstOrDefault(x => x.Count() > 1);

            if (duplicateType != null)
            {
                throw new ArgumentException($"Duplicate type in union: {duplicateType.Key}");
            }
```